### PR TITLE
Allow to deploy a network in idle mode. Activate later

### LINKED ISSUE #1532: Don't initialize StakingEscrow by default
I have serious concerns about our deployment process and how it affects day 0 of mainnet. I think we need to carefully design the process to minimize potential problems and blockages.

Currently, the `StakingEscrowDeployer` directly creates, funds, and initializes the `StakingEscrow` contract. This means that, by default, `StakingEscrow` is up and running once deployed. From that moment on, the clock is ticking and the first stakers have to create their stakes and confirm activity during the first 24 hours. But it can get worse: our current deployment process for `PreallocationEscrow` does something similar (i.e., creation, funding, and ownership transfer) and it can be quite intensive. On top of that, these contracts require knowing in advance the `StakingEscrow` address, so we can't deploy them before. It can be quite difficult to manage all those deployments in one day and still leave enough time for stakers to stake.

The good news is that the contracts are well-prepared for this. So, we only need to design carefully the deployment process and make the proper changes to the deployers and CLI, which I don't think it's too much work. It's just a matter of thinking ahead now.

My proposal:
- Don't initialize `StakingEscrow` by default. This means we can still do all the previous deployment steps, and leave initialization to when it's the right time. This also imply we need a CLI option to initialize an existing `StakingEscrow`.
- Deploy the preallocation contracts in the interim between `StakingEscrow` deployment and initialization. Transfer ownership of the contracts after the initialization (although perhaps there's no risk about doing that together with the deployment, after all `StakingEscrow` is not active yet. After #1407, it even doesn't have tokens yet!)
- Encourage stakers to set up their worker nodes before they can create the stake. Maybe there should be a way to create and run an "empty" Ursula that can run the discovery loop and checks that it's accessible.

### PR DESCRIPTION
**Pull request by cygnusv**
Closes #1532.

Recipe to deploy a network in idle mode:

1. Deploy a NuCypher contract

```
nucypher-deploy contracts --contract-name NuCypherToken --provider $NUCYPHER_PROVIDER_URI --poa --registry-infile ...
```
2. Deploy a StakingEscrow in IDLE mode
```
nucypher-deploy contracts --contract-name StakingEscrow --mode idle --provider $NUCYPHER_PROVIDER_URI --poa --registry-infile ...
```
3. Deploy PolicyManager and Adjudicator
```
nucypher-deploy contracts --contract-name PolicyManager --provider $NUCYPHER_PROVIDER_URI --poa --registry-infile ...
nucypher-deploy contracts --contract-name Adjudicator --provider $NUCYPHER_PROVIDER_URI --poa --registry-infile ...
```
4. When you're ready, activate StakingEscrow
```
nucypher-deploy contracts --contract-name StakingEscrow --activate --provider $NUCYPHER_PROVIDER_URI --poa --registry-infile ...
```

### CODE REVIEW

File: nucypher/cli/commands/deploy.py [hunk 1]
```diff
@@ -359,20 +366,27 @@ def rollback(general_config, actor_options):
...
 @group_general_config
 @group_actor_options
-@click.option('--bare', help="Deploy a contract *only* without any additional operations.", is_flag=True)
 @option_gas
 @option_ignore_deployed
 @click.option('--confirmations', help="Number of required block confirmations", type=click.IntRange(min=0))
 @click.option('--parameters', help="Filepath to a JSON file containing additional deployment parameters",
               type=EXISTING_READABLE_FILE)
-def contracts(general_config, actor_options, bare, gas, ignore_deployed, confirmations, parameters):
+@click.option('--mode',
```
Comment (KPrasch) [inline nucypher/cli/commands/deploy.py]: :+1:

File: nucypher/blockchain/eth/deployers.py [hunk 2]
```diff
@@ -54,13 +61,13 @@ class BaseContractDeployer:
...
     contract_name = NotImplemented
     deployment_steps = NotImplemented
-    upgrade_steps = NotImplemented
-    rollback_steps = NotImplemented
 
     _upgradeable = NotImplemented
     _ownable = NotImplemented
     _proxy_deployer = NotImplemented
 
+    can_be_idle = False
```
Comment (derekpierre) [inline nucypher/blockchain/eth/deployers.py]: Is this used somewhere?

File: nucypher/blockchain/eth/deployers.py [hunk 3]
```diff
@@ -654,22 +674,39 @@ def deploy(self,
...
-        deployment_receipts = dict(zip(self.deployment_steps, ordered_receipts))
+        activation_receipts = dict(zip(self.activation_steps, (approve_reward_receipt, init_receipt)))
+        return activation_receipts
 
-        # Set the contract and transaction receipts #
-        self._contract = the_escrow_contract
-        self.deployment_receipts = deployment_receipts
-        return deployment_receipts
+    @property
+    def ready_to_activate(self) -> bool:
```
Comment (vzotova) [inline nucypher/blockchain/eth/deployers.py]: I guess it's duplicated in the next method `is_active`, maybe remove this and keep other method?

File: nucypher/blockchain/eth/deployers.py [hunk 4]
```diff
@@ -654,22 +674,39 @@ def deploy(self,
...
+    def is_active(self) -> bool:
+        try:
+            deployed_contract = self._get_deployed_contract()
+        except self.blockchain.UnknownContract:
+            return False
+
+        # TODO: Consider looking for Initialized event - see #1193
+        # This mimics isInitialized() modifier in Issuer (StakingEscrow's base contract)
+        current_supply_1 = deployed_contract.functions.currentSupply1().call()
+        return current_supply_1 != 0
```
Comment (vzotova) [inline nucypher/blockchain/eth/deployers.py]: I'm for removing TODO and keeping check like now. If changes to `currentSupply1` is visible then block was added to chain

File: nucypher/blockchain/eth/deployers.py (hunk 3 above)
Comment (cygnusv) [reply nucypher/blockchain/eth/deployers.py]: Not really, this method deals with `current_supply_1 == 0`, while the next one is `current_supply_1 != 0`.

Comment (vzotova) [reply nucypher/blockchain/eth/deployers.py]: I see, this confuses, especially `ready_to_activate` and 
 when `current_supply_1 != 0` which means contract activated and never will be ready to activate again

**Pull request merged**